src/armor_layers.cpp

```diff
@@ -112,387 +106,481 @@ item_penalties get_item_penalties( std::list<item>::const_iterator worn_item_it,
     } else if( worn_item.has_flag( "WAIST" ) ) {
-        layer = _( "This is worn on or around your waist." );
+        layer = _( "(waist)" );
     } else if( worn_item.has_flag( "OUTER" ) ) {
-        layer = _( "This is worn over your other clothes." );
+        layer = _( "(outer)" );
     } else if( worn_item.has_flag( "BELTED" ) ) {
-        layer = _( "This is strapped onto you." );
+        layer = _( "(strap)" );
     } else if( worn_item.has_flag( "AURA" ) ) {
-        layer = _( "This is an aura around you." );
+        layer = _( "(aura)" );
     }
 
     return layer;
 }
-
-std::vector<std::string> clothing_properties(
-    const item &worn_item, const int width, const Character &c )
-{
-    std::vector<std::string> props;
-    props.reserve( 5 );
-
-    const std::string space = "  ";
-    props.push_back( string_format( "<color_c_green>[%s]</color>", _( "Properties" ) ) );
-    props.push_back( name_and_value( space + _( "Coverage:" ),
-                                     string_format( "%3d", worn_item.get_coverage() ), width ) );
-    props.push_back( name_and_value( space + _( "Encumbrance:" ),
-                                     string_format( "%3d", worn_item.get_encumber( c ) ),
-                                     width ) );
-    props.push_back( name_and_value( space + _( "Warmth:" ),
-                                     string_format( "%3d", worn_item.get_warmth() ), width ) );
-    props.push_back( name_and_value( space + string_format( _( "Storage (%s):" ), volume_units_abbr() ),
-                                     format_volume( worn_item.get_storage() ), width ) );
-    return props;
-}
-
-std::vector<std::string> clothing_protection( const item &worn_item, const int width )
+enum sort_armor_columns : int {
+    column_cond,
+    column_encumb,
+    column_storage,
+    column_warmth,
+    column_divider,
+    column_coverage,
+    column_bash,
+    column_cut,
+    column_acid,
+    column_fire,
+    column_env,
+    column_num_entries
+};
+struct sort_armor_col_data {
+    int width;
+    std::string name;
+    std::string description;
+};
+sort_armor_col_data get_col_data( int col_idx )
 {
-    std::vector<std::string> prot;
-    prot.reserve( 4 );
-
-    const std::string space = "  ";
-    prot.push_back( string_format( "<color_c_green>[%s]</color>", _( "Protection" ) ) );
-    prot.push_back( name_and_value( space + _( "Bash:" ),
-                                    string_format( "%3d", static_cast<int>( worn_item.bash_resist() ) ), width ) );
-    prot.push_back( name_and_value( space + _( "Cut:" ),
-                                    string_format( "%3d", static_cast<int>( worn_item.cut_resist() ) ), width ) );
-    prot.push_back( name_and_value( space + _( "Environmental:" ),
-                                    string_format( "%3d", static_cast<int>( worn_item.get_env_resist() ) ), width ) );
-    return prot;
+    static std::array< sort_armor_col_data, column_num_entries> col_data = { {
```

Comment:
Don't use static here, otherwise the translation won't update once loaded even if the language has been changed.